rfcs/0028-nix-release-model.md

```diff
@@ -0,0 +1,127 @@
+---
+feature: nix-release-model
+start-date: 2018-03-28
+author: Shea Levy <shea@shealevy.com>
+co-authors: (find a buddy later to help our with the RFC)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+A model of when, what, and how to make new releases of Nix.
+
+# Motivation
+[motivation]: #motivation
+
+We want to set clear expectations for users, provide guidance for
+developers, and provide both groups the benefits of easy availability
+of new improvements.
+
+# Detailed design
+[design]: #detailed-design
+
+The proposed process contains a number of interrelated components.
+I've numbered them for ease of reference and discussion, not to
+indicate any meaningful ordering. Any aspect of this can be suspended
+at the security team's discretion when handling a vulnerability. Items
+marked with a * need technical work to implement.
+
+1. Keep `master` as close to release readiness as possible. This
+   includes testing, documentation, changelog entries, etc. being
+   ready *before* merge.
+2. *Use code review and automated checks, enforced through branch
+    protection, to ensure point 1 (and generally improve quality and
+    increase knowledge spread).
+3. Follow semver. In light of 1, this includes ensuring any necessary
+   semver bumps happen before a relevant change is merged.
+4. Any community member may request a new release by opening an issue
+   on GitHub. If any Nix core team member (who isn't the requestor)
+   thinks a release is warranted, they can assign themselves the
+   ticket and assume release manager responsibilities for the release.
+5. There will be at least one mainline release before each `NixOS`
+   stable release.
```

Comment:
I think this implies a weaker form of 4 to be useful. If no core team member is willing to initiate a release, maybe there shouldn't be one even if the specified number of months has elapsed?

Reply:
I think we should view this as a commitment by the core team that *someone* will be willing to step up. If development on Nix really slows down that much we can change this.

Reply:
If the question is someone stepping up, maybe add a provision that as a normal Nix core team decision (by the +2 -1 rule) a willing Nixpkgs committer can be granted release responsibilities?

I just want to say that if this point is ever relevant with the process as described, there is probably an unanimous agreement that it is hard/risky/pointless to do release right now (if the only problem is finding time, asking for help sounds like a good idea).

Reply:
OK, will clarify.